x-pack/filebeat/input/httpjson/config.go

```diff
@@ -84,6 +94,11 @@ func (c *config) Validate() error {
 			}
 		}
 	}
+	if c.OAuth2 != nil {
```

Comment:
Maybe others might think differently, but maybe nice to have a enable/disable flag like pagination has, that way they could easily disable it on the fly without having to remove all configuration

Reply:
If we are still looking for this being nil or not, then having enabled/disable option will still throw an error I believe? Maybe add a second condition saying "if its not null and not disabled"?

Reply:
@andrewkroh pointed out the same, WDYT about this? https://github.com/elastic/beats/pull/18892#discussion_r435727095